zerver/views/video_calls.py

```diff
@@ -161,18 +161,29 @@ def deauthorize_zoom_user(request: HttpRequest) -> HttpResponse:
     return json_success()
 
 
-def get_bigbluebutton_url(request: HttpRequest, user_profile: UserProfile) -> HttpResponse:
+@has_request_variables
+def get_bigbluebutton_url(request: HttpRequest, user_profile: UserProfile,
+                          meeting_name: str = REQ()) -> HttpResponse:
     # https://docs.bigbluebutton.org/dev/api.html#create for reference on the api calls
     # https://docs.bigbluebutton.org/dev/api.html#usage for reference for checksum
     id = "zulip-" + str(random.randint(100000000000, 999999999999))
     password = b32encode(secrets.token_bytes(7))[:10].decode()
-    checksum = hashlib.sha1(("create" + "meetingID=" + id + "&moderatorPW="
-                             + password + "&attendeePW=" + password + "a" + settings.BIG_BLUE_BUTTON_SECRET).encode()).hexdigest()
-    url = add_query_to_redirect_url("/calls/bigbluebutton/join", urlencode({
-        "meeting_id": "\"" + id + "\"",
-        "password": "\"" + password + "\"",
-        "checksum": "\"" + checksum + "\""
-    }))
+    create_params = urlencode({  # type: ignore[type-var] # https://github.com/python/typeshed/issues/4234
+        "meetingID": id,
+        "name": meeting_name,
+        "moderatorPW": password,
+        # We generate the password this way to make sure only the server knows it.
+        # This is a easy way to prevent the user to get the data of the shared url
+        # and use it later and skipping Zulip validation
+        "attendeePW": hashlib.sha1((password + settings.BIG_BLUE_BUTTON_SECRET).encode()).hexdigest(),
```

Comment:
We should probably use sha2 here, since sha1 is insecure?

Reply:
At most places I have to use sha1, because the Big Blue Button API requires it for the checksum.

Fo this attendePW thing I do not have to use sha1 and will either change this to sha2 or find a better solution (I have one in mind)

Reply:
changed in `verify url data` commit